lisa/sut_orchestrator/azure/common.py

```diff
@@ -1212,9 +1222,16 @@ def load_environment(
         ) = get_primary_ip_addresses(
             platform, resource_group_name, vms_map[node_context.vm_name]
         )
+
+        connection_public_ip = node_context.public_ip_address
+        if use_private_ip:
+            # setting public_address to empty causes set_connection_info
+            # to use address (the private ip)
+            connection_public_ip = ''
```

Comment:
might make more sense to change this to use `node_context.private_ip_address` rather than an empty string.